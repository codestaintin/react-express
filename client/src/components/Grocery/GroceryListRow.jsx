import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const GroceryListRow = ({
  grocery, handleDelete, handlePurchase
}) => (
  <Fragment>
    <tr>
      <th scope="row">&nbsp;</th>
      <td>{grocery.name}</td>
      <td>
        <button
          type="button"
          className={`btn ${grocery.purchased ? 'btn-info' : 'btn-primary'}`}
          onClick={() => handlePurchase(grocery._id)}
        >
          {grocery.purchased ? 'Bought' : 'Buy'}
        </button>
      </td>
      <td>
        <button
          type="button"
          className="btn btn-danger"
          onClick={() => handleDelete(grocery._id)}
        >
              Delete Item
        </button>
      </td>
    </tr>
  </Fragment>
);

GroceryListRow.propTypes = {
  grocery: PropTypes.shape({}),
  handleDelete: PropTypes.func.isRequired,
  handlePurchase: PropTypes.func.isRequired
};

export default GroceryListRow;
