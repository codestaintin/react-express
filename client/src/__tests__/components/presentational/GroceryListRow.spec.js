import React from 'react';
import { shallow } from 'enzyme';
import GroceryListRow from '../../../components/Grocery/GroceryListRow';

describe('<GroceryListRow>', () => {
  const props = {
    grocery: {},
    handleDelete: jest.fn(() => {}),
    handlePurchase: jest.fn(() => {})
  };

  it('should render GroceryListRow', () => {
    const tree = shallow(<GroceryListRow {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('simulate a buy click', () => {
    const tree = shallow(<GroceryListRow {...props} />);
    tree.find('.btn-primary').simulate('click');
  });

  it('simulate a delete click', () => {
    const tree = shallow(<GroceryListRow {...props} />);
    tree.find('.btn-danger').simulate('click');
  });
});
