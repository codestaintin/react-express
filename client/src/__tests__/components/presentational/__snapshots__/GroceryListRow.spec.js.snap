// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroceryListRow> should render GroceryListRow 1`] = `
<Fragment>
  <tr>
    <th
      scope="row"
    >
      Â 
    </th>
    <td />
    <td>
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Buy
      </button>
    </td>
    <td>
      <button
        className="btn btn-danger"
        onClick={[Function]}
        type="button"
      >
        Delete Item
      </button>
    </td>
  </tr>
</Fragment>
`;
